### PR TITLE
add editable component

### PR DESCRIPTION
**Pull request by omair2010**
## Description 
https://github.com/BearStudio/start-ui/issues/31

Add code editable text area component

## Demo
https://user-images.githubusercontent.com/32815467/107668665-33cc0b00-6c91-11eb-97b4-3a59322657a9.mp4

### CODE REVIEW

File: src/components/Editable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+              {item}
+              <br />
+            </Fragment>
+          ))}
+        </Text>
+      )}
+      <ButtonGroup size="sm">
+        {isEditing && (
+          <IconButton
+            aria-label={t('components.editable.cancel')}
```
Comment (HugoPerard) [inline src/components/Editable/index.tsx:94]: Remove i18n keys, write text directly

File: src/components/Editable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            onClick={handleCancel}
+            icon={<HiX />}
+          />
+        )}
+        <IconButton
+          aria-label={
+            isEditing
+              ? t('components.editable.submit')
+              : t('components.editable.edit')
```
Comment (HugoPerard) [inline src/components/Editable/index.tsx:103]: Remove i18n keys, write text directly

File: src/components/Editable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          transition="none"
+          ml="-0.5rem"
+          mr={3}
+          px={2}
+          py={1}
+          minH={4}
+          autoFocus
+          maxRows={10}
+          onChange={handleChange}
+        />
```
Comment (HugoPerard) [inline src/components/Editable/index.tsx:74]: Maybe you should extract TextareaAutosize component, all these properties are required ?

**Review by HugoPerard — CHANGES_REQUESTED** [review]
Component do not work, text should be staying visible in non-editing state

File: src/components/TextareaAutosize/Editable.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { TextareaAutosize } from '.';
```
Comment (HugoPerard) [inline src/components/TextareaAutosize/Editable.stories.tsx:1]: Wrong file name

File: src/components/TextareaAutosize/index.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+export const TextareaAutosize: FC<TextareaAutosizeProps> = (props) => (
+  <Textarea
+    as={TextareaAutosizeReact}
+    transition="none"
+    ml="-0.5rem"
+    mr={3}
+    px={2}
+    py={1}
+    minH={4}
+    autoFocus
```
Comment (HugoPerard) [inline src/components/TextareaAutosize/index.tsx:17]: These properties (maybe not minH) should be only on the TextareaAutosize of the Editable

File: src/components/Editable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+} from '@chakra-ui/react';
+import { HiCheck, HiPencilAlt, HiX } from 'react-icons/hi';
+
+import { TextareaAutosize } from '../TextareaAutosize';
+
+export interface EditableProps extends FlexProps {
+  value?: string;
+  onSubmit?: (string) => void;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onCancel?: (string) => void;
```
[the remark this answers is not in the document]
Comment (omair2010) [reply src/components/Editable/index.tsx:18]: we need argument

**Pull request closed without merging**